### ISSUE TITLE
feature request: Uniques createAndMint or createAndSetMetadata / mintAndSetMetadata

### ISSUE THREAD

**Issue by Yuripetusko** [paritytech/substrate#9778]
When creating a uniques class class, and then adding metadata with setClassMetadata on that class all in one tx using batch transaction I received an error. It looks like I need to have 2 separate transactions perhaps in separate blocks, so class needs to be created first and tx approved and only then I can add metadata to it.

This creates a bad user experience that to create a class user needs to sign 2 transactions, what is even worse is that to create a first NFT user needs to sign 4 transactions. (2 for class and it's metadata and 2 for asset and it's metadata)

So it would be great to: a) have a way to pass metadata at a point of class creation or asset mint and b) have a single call to create a class and mint a transaction all in one tx. b) is lower priority here I would say for UX point of view as user usually needs to create a class only once. But mint and setting metadata would be great to be able to do in 1 call

**Comment by hamidra** [comment]
I just run a batch[create, setClassMetadata, mint] and it worked with no error. Can you share the error you were getting when you were running the batch?!

**Comment by hamidra** [comment]
Anyhow the error seems to be a user error in the transaction since the batch will run in order and all combination of calls, as long as they are batched in correct order, should work as expected. I'll close this. feel free to reopen with more info if you see this is not behaving as expected.

**Comment by Yuripetusko** [comment]
@hamidra If this is indeed my client side error, then I apologize for wasting your time. Will verify this in next 2 days and update here. Really appreciate you looking into this, and sorry again

**Comment by Yuripetusko** [comment]
This now works `[create, setClassMetadata, mint, setMetadata]` this was human error on my part, I also wasn't aware that uniques class require a deposit of 1KSM
